source/common/event/scaled_range_timer_manager.cc

```diff
@@ -17,16 +17,25 @@ namespace Event {
  * Implementation of RangeTimer that can be scaled by the backing manager object.
  *
  * Instances of this class exist in one of 3 states:
- *  - disabled: not enabled
+ *  - inactive: not enabled
  *  - waiting-for-min: enabled, min timeout not elapsed
  *  - scaling-max: enabled, min timeout elapsed, max timeout not elapsed
+ *
+ * The allowed state transitions are:
+ *  - inactive -> waiting-for-min
+ *  - waiting-for-min -> scaling-max | inactive
+ *  - scaling-max -> inactive
+ *
+ * Some methods combine multiple state transitions; enableTimer(0, max) on a
+ * timer in the scaling-max state will logically execute the transition sequence
+ * [scaling-max -> inactive -> waiting-for-min -> scaling-max] in a single
```

Comment:
Where does the transition from scaling-max to inactive come from?

Reply:
If the timer is already enabled, the first step is to disable it. Then we can enable it again for the new duration.

Reply:
I missed that the timer described in the comment is initially in the scaling-max state.  The example may have been easier to follow if the initial state had been waiting-for-min